lib/index.js

```diff
@@ -295,8 +310,162 @@ Segment.prototype.referrerId = function(query, ctx) {
   this.cookie('s:context.referrer', json.stringify(ad));
 };
 
+
 /**
- * Noop.
+ * retrieveCrossDomainId.
+ *
+ * @api private
+ * @param {function) callback => err, {crossDomainId, fromServer, timestamp}
+ */
+Segment.prototype.retrieveCrossDomainId = function(callback) {
+  if (!this.options.crossDomainIdServers) {
+    if (callback) {
+      callback('crossDomainId not enabled', null);
+    }
+    return;
+  }
+  if (!this.cookie('segment_cross_domain_id')) {
+    var self = this;
+    var writeKey = this.options.apiKey;
+    
+    // Exclude the current domain from the list of servers we're querying
+    var currentTld = getTld(window.location.hostname);
```

Comment:
Will this be consistent with the TLD we get from segmentio/top-domain? I don't remember why we use that library off the top of my head, but @yields likely knows